### PR TITLE
feat(auth): implement login screen with Google Sign-In

### LINKED ISSUE #273: Login vertical
## Phase 1 — Parallel Track 1

Parent: #126 · Blocked by: #127 + #128 · Supersedes: #132, #133, #134

## Context

Complete vertical delivery of the login flow: use cases, credentials validator implementation, ViewModel, screen, and tests. This merges the previously separate use case (#132), ViewModel (#133), and screen (#134) issues into a single vertical unit per the Vertical Feature Delivery principle.

Supports both email/password and Google Sign-In login methods as decided in #275. Develops against fakes from #128 — no dependency on the real data layer (#131).

### Domain
- `LoginWithCredentialsUseCase` — validates credentials via `CredentialsValidator`, calls `AuthRepository.loginWithCredentials()`
  - On validation failure: return validation errors without network call
  - On success: auth state transitions to `Authenticated`
  - On failure: surface domain error (InvalidCredentials, NetworkError, Unknown)
- `LoginWithGoogleUseCase` — calls `AuthRepository.loginWithGoogle(idToken)`
  - On success: auth state transitions to `Authenticated`
  - On failure: surface domain error (InvalidToken, AccountNotFound, NetworkError, Unknown)
- `CredentialsValidatorImpl` — implements `CredentialsValidator` from #127
  - Email: not blank, valid email format
  - Password: not blank, minimum length

### UI
- `LoginViewModel` (MVI pattern)
  - State: email, password, loading, field errors, general error
  - Events: navigate to chat list on success
  - Actions: update fields, submit credential login, initiate Google Sign-In, navigate to signup
- `LoginScreen` (Jetpack Compose)
  - Email field, password field
  - Login button with loading state
  - Google Sign-In button
  - "Don't have an account? Sign up" link
  - Inline validation error display
  - General error display (invalid credentials, network error)

### Google Sign-In integration
- Google Sign-In SDK setup (Credential Manager API)
- Obtain Google ID token from sign-in result
- Pass ID token to `LoginWithGoogleUseCase`

### Navigation
- Navigate to chat list on successful login (clear auth back stack)
- Navigate to signup screen

### Tests
- `CredentialsValidatorImpl` unit tests (all validation rules)
- `LoginWithCredentialsUseCase` unit tests (validation failure, success, invalid credentials, network error)
- `LoginWithGoogleUseCase` unit tests (success, invalid token, account not found, network error)
- `LoginViewModel` unit tests (field updates, submission states, Google sign-in flow, navigation events)
- All tests use `AuthRepositoryFake` and `CredentialsValidatorStub`

## Out of scope
- `AuthRepositoryImpl` (belongs to #131)
- Signup flow (belongs to #232)
- Token refresh (belongs to #233)

## Acceptance criteria
- [ ] `CredentialsValidatorImpl` validates email and password rules
- [ ] `LoginWithCredentialsUseCase` validates before calling repository
- [ ] `LoginWithGoogleUseCase` passes Google ID token to repository
- [ ] `LoginViewModel` follows MVI pattern with single `StateFlow` + `Channel` events
- [ ] `LoginScreen` renders email/password fields, login button, and Google Sign-In button
- [ ] Invalid credentials error displayed from server response
- [ ] Google Sign-In errors handled and displayed
- [ ] Successful login (either method) navigates to chat list
- [ ] All tests pass using fakes from #128

### PR DESCRIPTION
**Pull request by timurgilfanov**
Closes #273

## Summary
Implements the complete Login screen as a vertical feature slice: UI, ViewModel, use cases, credentials validator, Hilt DI wiring, CredentialManager integration for Google Sign-In, and full unit test coverage.

## Changes
- **Use cases** — `LoginWithCredentialsUseCase` and `LoginWithGoogleUseCase` (`fun interface`) with sealed error hierarchies (`LoginUseCaseError`, `GoogleLoginUseCaseError`) and repository-to-use-case error mappings
- **Implementations** — `LoginWithCredentialsUseCaseImpl` (validates credentials, calls auth repository) and `LoginWithGoogleUseCaseImpl`
- **Validator** — `CredentialsValidatorImpl` with all email and password rules
- **ViewModel** — `LoginViewModel` (MVI: single `StateFlow` + `Channel`-backed effects) handling email/password login, Google Sign-In, and all error states
- **UI** — `LoginScreen` with email field, password field, inline validation errors, general error display, loading indicator, Sign In and Sign In with Google buttons, and Create Account navigation
- **CredentialManager** — `LoginScreen` builds the `CredentialManager` request internally; cancellation is silent, infrastructure errors surface as `LoginGeneralError.GoogleSignInFailed`; ViewModel receives only the resulting ID token string
- **Strings** — English and German string resources for all labels, validation errors, general errors, and `google_server_client_id` placeholder
- **DI** — `AuthModule` binds use case interfaces to their implementations
- **Navigation** — `Login` destination added; `MainActivity` wires `LoginScreen` into the nav graph; old stub `LoginScreen` in `ui.screen.settings` removed
- **Dependencies** — `androidx.credentials`, `credentials-play-services-auth`, and `googleid` added to version catalog and `feature/auth`
- **Tests** — `LoginViewModelInitialStateTest`, `LoginViewModelSubmitTest`, `LoginViewModelGoogleSignInTest`, `LoginWithCredentialsUseCaseTest`, `LoginWithGoogleUseCaseTest`, `CredentialsValidatorImplTest`

## Important Design Decision
`CredentialManager` requires an `Activity` context, so credential acquisition belongs at the screen level. The ViewModel stays infrastructure-agnostic — it receives only the ID token string — which keeps it fully unit-testable without Android mocks.

## Testing
- [x] Unit tests added or updated
- [ ] Manually tested
- [x] Edge cases considered

## Screenshots / Demo
image
image

_Login screen previews._

## Acceptance criteria
- [x] `CredentialsValidatorImpl` validates email and password rules
- [x] `LoginWithCredentialsUseCase` validates before calling repository
- [x] `LoginWithGoogleUseCase` passes Google ID token to repository
- [x] `LoginViewModel` follows MVI pattern with single `StateFlow` + `Channel` events
- [x] `LoginScreen` renders email/password fields, login button, and Google Sign-In button
- [x] Invalid credentials error displayed from server response
- [x] Google Sign-In errors handled and displayed
- [x] Successful login (either method) navigates to chat list
- [x] All tests pass using fakes from #128

## Breaking Changes
None — `LoginScreen` is a new screen. The old `ui.screen.settings.LoginScreen` stub is removed (it was unreachable).

## Follow-ups
- [x] Replace `google_server_client_id` placeholder with the real Web OAuth client ID once available (#302)
- [x] Wire real auth repository backend when available (#131)
- [x] Split `CredentialsValidationError` into `EmailValidationError` and `PasswordValidationError` (#304)

### CODE REVIEW

File: feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/login/LoginScreen.kt [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+        snackbarHostState = snackbarHostState,
+        onEmailChange = viewModel::updateEmail,
+        onPasswordChange = viewModel::updatePassword,
+        onSubmitLogin = viewModel::submitLogin,
+        onGoogleSignInClick = {
+            scope.launch {
+                when (val result = googleSignInClient.signIn(context)) {
+                    is GoogleSignInResult.Success -> viewModel.submitGoogleSignIn(result.idToken)
+                    GoogleSignInResult.Cancelled -> Unit
+                    GoogleSignInResult.Failed -> viewModel.onGoogleSignInFailed()
```
Comment (timurgilfanov) [inline feature/auth/src/main/kotlin/timur/gilfanov/messenger/auth/login/LoginScreen.kt]: We have no reasons to call VM because effect is UI-only. We could show snackbar without VM
```suggestion
                    GoogleSignInResult.Failed -> snackbarHostState.showSnackbar(
                            message = stringResourse(R.string.login_error_google_sign_in_failed),
                            duration = SnackbarDuration.Long,
                        )
```
